app/assets/javascripts/controllers/FileCtrl.coffee

```diff
@@ -0,0 +1,41 @@
+@bruseApp.controller 'FileCtrl', ['FileHandler', '$scope', '$http', (FileHandler, $scope, $http) ->
+  $scope.files = []
+  $scope.loading = true
+
+  # get identity information from add view
+  $scope.identity = IDENTITY_PARAMS
+
+  # load root files on page load
+  FileHandler.get($scope.identity.id, '/').then((data) ->
+    $scope.files = data
+    $scope.loading = false
+    )
+
+  $scope.expand = (file) ->
+    # set this folder as expanded
+    file.expand = !file.expand
+
+    # if we haven't allready, load file info from dropbox
+    unless file.contents and file.contents.length > 0
+      $scope.loading = true
+      FileHandler.get($scope.identity.id, file.path).then((data) ->
+        file.contents = data
+        $scope.loading = false
+        )
+
+  $scope.add = (file) ->
+    # make sure we are not adding a file
+    unless file.is_dir
```

Comment:
`is_dir` is a propertry that we get from Dropbox. It's a boolean that says if the object is a folder or if it's actually a file. On this line we make sure that the current object actually is a file, and *not* a folder. 

I guess it would be nice to be able to add entire folders to, but that requires further work.